build/mac/pkg-scripts/postinstall

```diff
@@ -0,0 +1,52 @@
+#!/bin/sh
+# This postinstall script is run on macOS pkg installers
+
+# Extract a referral / promo code from the installer path of the format /dir/path/Setup-Brave-Anything-xxx001.pkg
+# where the promo code would be xxx001
+installerPath=$1
+installationPath=$2
+
+# TODO: ugly to assume the name of the app, especially with multi-channel.
+# Luckily for now, release channel is the only channel with a pkg installer.
+installationAppPath="$installationPath/Brave Browser.app"
+installerPathPromoCodeRegex='.+-(([a-zA-Z0-9]{3}[0-9]{3})|([a-zA-Z]{1,}-[a-zA-Z]{1,}))([[:blank:]]?\([0-9]+\))?\.pkg$'
+userDataDir="$HOME/Library/Application Support/BraveSoftware/Brave-Browser"
+promoCodePath="$userDataDir/promoCode"
+
+# pkg runs this script as root so we need to get current username
+userName="$USER"
+
+echo "Installer path is: $installerPath"
+echo "Username is: $userName"
+echo "Installation app path is: $installationAppPath"
+
+# Fix the permissions on installed .app so that updater has permissions to write new contents.
+# By default pkg will install the .app with root:wheel owner and 'drwxr-xr-x' permission.
+# We'll allow all admin users of the machine permissions to update the app, as well as the installing-user
+# (who may not be an admin).
+sudo chmod -R 775 "$installationAppPath"
+sudo chown -R $userName "$installationAppPath"
+sudo chgrp -R admin "$installationAppPath"
```

Comment:
[Security review] How does this compare to Chrome's directory permissions? In general we want to match that (or be more restrictive)

Reply:
This is how Chrome installs in my `/Applications` folder on Mac:

```
$ ls -l | grep Chrome
drwxrwxr-x@  3 emerick  admin     96 Sep 15 04:30 Google Chrome.app/
```

So I think we're exactly matching the permissions/owner/group of Chrome.

Reply:
That is correct, as far as I know- @petemill did this and I remember digging in with him. We wanted to match Chrome